src/promptflow/promptflow/_sdk/schemas/_flow.py

```diff
@@ -58,6 +63,22 @@ class EagerFlowSchema(BaseFlowSchema):
     """Schema for eager flow."""
 
     # path to flow entry file.
-    path = LocalPathField(required=True)
+    path = LocalPathField(required=False)
     # entry function
     entry = fields.Str(required=True)
+
+    @post_load
+    def infer_path(self, data: dict, **kwargs):
+        """Infer path from entry."""
+        # TODO: remove this after path is removed
+        language = data.get(LANGUAGE_KEY, FlowLanguage.Python)
+        if language == FlowLanguage.Python and data.get("path", None) is None:
+            raise ValidationError(message={"path": ["Missing data for required field."]})
+        elif language == FlowLanguage.CSharp and data.get("path", None) is None:
+            # for csharp, path to flow entry file will be a dll path inferred from
+            # entry by default given customer won't see the dll on authoring
+            m = re.match(r"\((.+)\)[a-zA-Z0-9]+(\.[a-zA-Z0-9]+)+", data["entry"])
+            if not m:
+                raise ValueError(f"Invalid entry: {data['entry']}")
```

Comment:
Why not validation error. @Stephen1993 what's the error category for value error?